app/models/interaction.rb

```diff
@@ -0,0 +1,3 @@
+class Interaction < ActiveRecord::Base
+  validates :identifier, :label, presence: true, uniqueness: true
+end
```

Comment:
Now that we've got uniqueness and presence validations it's worth backing those up with not-null and uniqueness constraints at the database level in our migrations.  [This thoughtbot article](https://robots.thoughtbot.com/the-perils-of-uniqueness-validations) is a pretty good discussion about why you might want to add db constraints as well as doing application-level validation.